pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -392,62 +392,66 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 	return volumeName, nil
 }
 
-func (dsw *desiredStateOfWorld) getSELinuxLabel(volumeSpec *volume.Spec, seLinuxContainerContexts []*v1.SELinuxOptions) (string, bool, error) {
-	var seLinuxFileLabel string
-	var pluginSupportsSELinuxContextMount bool
+func (dsw *desiredStateOfWorld) getSELinuxLabel(volumeSpec *volume.Spec, seLinuxContainerContexts []*v1.SELinuxOptions, podSecurityContext *v1.PodSecurityContext) (string, bool, error) {
+	if !feature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) {
+		return "", false, nil
+	}
 
-	if feature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) {
-		var err error
+	if !dsw.seLinuxTranslator.SELinuxEnabled() {
+		return "", false, nil
+	}
 
-		if !dsw.seLinuxTranslator.SELinuxEnabled() {
-			return "", false, nil
-		}
+	pluginSupportsSELinuxContextMount, err := dsw.getSELinuxMountSupport(volumeSpec)
+	if err != nil {
+		return "", false, err
+	}
 
-		pluginSupportsSELinuxContextMount, err = dsw.getSELinuxMountSupport(volumeSpec)
+	if feature.DefaultFeatureGate.Enabled(features.SELinuxChangePolicy) &&
+		podSecurityContext != nil &&
+		podSecurityContext.SELinuxChangePolicy != nil &&
+		*podSecurityContext.SELinuxChangePolicy == v1.SELinuxChangePolicyRecursive {
+		// The pod has opted into recursive SELinux label changes. Do not mount with -o context.
+		return "", pluginSupportsSELinuxContextMount, nil
+	}
+
+	if !pluginSupportsSELinuxContextMount {
+		return "", pluginSupportsSELinuxContextMount, nil
+	}
+
+	seLinuxSupported := util.VolumeSupportsSELinuxMount(volumeSpec)
```

Comment:
Why does `VolumeSupportsSELinuxMount` return false for volumes without access modes?

Reply:
Good catch! That's leftover of RWOP, which needed to be the only item in AccessModes.